tests/protocols/analytics_test.py

```diff
@@ -0,0 +1,58 @@
+from client.protocols import analytics
+from os import path
+
+import mock
+import unittest
+
+class TestAnalyticsProtocol(unittest.TestCase):
+    def setUp(self):
+        self.cmd_args = mock.Mock()
+        self.cmd_args.submit = True
+        self.assignment = mock.MagicMock()
+        self.file_dir = path.abspath(path.join(path.dirname(__file__), '..', "data"))
+
+        self.proto = analytics.protocol(self.cmd_args, self.assignment)
+
+    def read_file(self, file_name):
```

Comment:
Instead of actually performing I/O operations, you should just put the "file contents" as strings in this test file.